app/components/searchworks4/volumes_modal_component.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Searchworks4
+  class VolumesModalComponent < ViewComponent::Base
+    def initialize(call_number:, response:, title:)
+      super()
+      @call_number = call_number
+      @response = response
+      @title = title
+    end
+
+    def results
+      @response.dig(:response, :docs)
+    end
+
+    def call_number_display(solr_doc)
+      if solr_doc.key?('lc_assigned_callnum_ssim')
+        solr_doc['lc_assigned_callnum_ssim'].join(', ')
+      elsif solr_doc.key?('item_display_struct')
+        solr_doc['item_display_struct'][0]['callnumber']
+      else
+        ''
```

Comment:
I don't think this is the call number logic we want to use. Maybe there's existing code for browse nearby that does it correctly? (At the very least, using the LC assigned call number by default is giving us bad call number data when our local call numbers don't match (e.g. we're using Dewey or something)

Reply:
I can look into browse nearby. I was quite unsure about how call numbers are supposed to work here so was hoping for some feedback on this part.